### DISCUSSION TITLE
How to read NAV-RELPOSNED or NAV-HPPOSLLH.

### DISCUSSION THREAD

**Discussion by Sebtronico** [semuconsulting/pyubx2#91] [Q&A] [answered]
I want to read any of these registers to know the position of the device configured as Rover, but I don't know how to do it.

**Comment by semuadmin** [comment, marked as answer]
Hi @Sebtronico 

Apologies for the belated response. You can decode NAV-RELPOSNED and NAV-HPPOSLLH message types using PyGPSClient or pyubx2. You'll need access to the serial stream on your roving device (or set it up to transmit this stream over a TCP socket instead). The position can then be obtained from the UBXMessage `lat` and `lon` attributes.
